### ISSUE TITLE
accounts: Redesign password status & actions

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#14725]
Accounts page has an action hidden behind a link that is overloaded as a status message.

Instead, the password part of the user account page should:
- have a summary including:
  - when the password was set
  - when the password will expire, including:
    - expiration date
    - will not expire
    - will expire on next log in
- actions as buttons, with the functionality of:
   - changing the password to a predefined string the administrator sets
   - requiring the user to set the password on the next login (which may have a placeholder string or may just use the current password)
   - requiring the user to change the password at intervals (every X days, every X weeks, etc.)
   - _possibly_ requiring the user to change the password at X days from the current time

Most of this already exists in the current interface. The big change is the split of the status and actions. Actions such as force change and expiration should probably be grouped together, possibly with the differences between them set within a modal.

**Comment by marusak** [comment]
TL;DR: This issue is fixed. There are some possible improvements on the table, but I believe that it could do with overall redesign and fix it while on that.

Since https://github.com/cockpit-project/cockpit/pull/15832 we don't overload status as a link. Now it looks somewhat like this:
![Screenshot from 2022-03-04 09-11-44](https://user-images.githubusercontent.com/12330670/156725329-ec9c2d92-c482-4423-8235-26b80e6146ea.png)

I don;t have previous version built, but when this was opened, there would not be any `edit` just the status e.g. `Expire on ...` would be blue and behave as now does `edit`.

So the main idea of this is implemented. Surely the UI can do with some improvements, but not sure if that is goal of this issue.

Above points:

- [ ] when the password was set
- when the password will expire, including:
    - [x] expiration date
    - [x] will not expire
    - [x] will expire on next log in
- actions as buttons, with the functionality of:
    - [x] changing the password to a predefined string the administrator sets
    - [x] requiring the user to set the password on the next login (which may have a placeholder string or may just use the current password)
    - [x] requiring the user to change the password at intervals (every X days, every X weeks, etc.)
    - [x] possibly requiring the user to change the password at X days from the current time

> Most of this already exists in the current interface. The big change is the split of the status and actions. Actions such as force change and expiration should probably be grouped together, possibly with the differences between them set within a modal.

It indeed already exists. It also is split into status and action. Agree that force change and expiration could maybe be one dialog - but that needs some good design.

The only thing that is missing is when the password was changed. I am not sure if we can figure this information out or not. But also, not sure where to put it into UI.

@garrett WDYT? Do you agree that the main UX concern was already addressed and we can close this and that we should do another issue/jira card to track redesign?

**Comment by marusak** [comment]
Redesign is happening therefore closing this
